src/lib.rs

```diff
@@ -254,68 +257,43 @@ impl Pty {
     })
   }
 
-  /// Resize the terminal.
+  /// Transfers ownership of the file descriptor for the PTY controller. This can only be called
+  /// once (it will error the second time). The caller is responsible for closing the file
+  /// descriptor.
   #[napi]
   #[allow(dead_code)]
-  pub fn resize(&mut self, size: Size) -> Result<(), napi::Error> {
-    let window_size = Winsize {
-      ws_col: size.cols,
-      ws_row: size.rows,
-      ws_xpixel: 0,
-      ws_ypixel: 0,
-    };
-
-    if let Some(fd) = &self.controller_fd {
-      let res = unsafe { libc::ioctl(fd.as_raw_fd(), libc::TIOCSWINSZ, &window_size as *const _) };
-      if res == -1 {
-        return Err(napi::Error::new(
-          napi::Status::GenericFailure,
-          format!("ioctl TIOCSWINSZ failed: {}", Error::last_os_error()),
-        ));
-      }
-
-      Ok(())
-    } else {
-      Err(napi::Error::new(
-        napi::Status::GenericFailure,
-        "ioctl TIOCSWINSZ failed: bad file descriptor (os error 9)",
-      ))
-    }
-  }
-
-  /// Returns a file descriptor for the PTY controller.
-  /// See the docstring of the class for an usage example.
-  #[napi]
-  #[allow(dead_code)]
-  pub fn fd(&mut self) -> Result<c_int, napi::Error> {
-    if let Some(fd) = &self.controller_fd {
-      Ok(fd.as_raw_fd())
+  pub fn take_fd(&mut self) -> Result<c_int, napi::Error> {
+    if let Some(fd) = self.controller_fd.take() {
```

Comment:
is it fine that we take here? doesn't poll_controller_fd_until_read have an unsafe ref to this?

just wondering if theres a case where node exits and closes the fd and then we try to poll on it

Reply:
i guess the `break` from `EAGAIN` or `EINTR` would cover that

Reply:
yup, that's totally fine. we don't actually _read_ from it, just check that it's readable. and in the w.c.s. we will exit 1s later.